confmap/provider/internal/configurablehttpprovider/provider.go

```diff
@@ -0,0 +1,194 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package configurablehttpprovider // import "github.com/Dynatrace/dynatrace-otel-collector/confmap/provider/internal/configurablehttpprovider"
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+type SchemeType string
+
+const (
+	// "EEC Insecure" - equivalent to HTTP
+	EECIScheme SchemeType = "eeci"
+	// equivalent to HTTPS
+	EECScheme SchemeType = "eec"
```

Comment:
I went with a "secure by default" approach here with having the extra letter indicate an insecure connection instead of being required to use TLS.

I realize that this goes against the "HTTP"/"HTTPS" convention, so if we don't think this is worth it, I can update this to be something like "eec"/"eecs".

Reply:
To be honest, I was confused the first time I saw the `eeci` SchemeType, as I assumed it should be `eec` / `eecs` cause your given reason.

I don't have a strong opinion on that but would be slightly in favor of `eecs` for old grumpy admins like I am ;)